modules/extensions-xml-support/src/main/java/org/mule/runtime/extension/internal/loader/enricher/StereotypesDiscoveryDeclarationEnricher.java

```diff
@@ -0,0 +1,178 @@
+ * used extensions adding the needed {@link StereotypeModel} to the current declaration.
+ *
+ * @since 4.2
+ */
+public class StereotypesDiscoveryDeclarationEnricher implements DeclarationEnricher {
+
+  @Override
+  public DeclarationEnricherPhase getExecutionPhase() {
+    return POST_STRUCTURE;
+  }
+
+  @Override
+  public void enrich(ExtensionLoadingContext extensionLoadingContext) {
+    withContextClassLoader(extensionLoadingContext.getExtensionClassLoader(),
+                           () -> new EnricherDelegate(extensionLoadingContext.getDslResolvingContext())
+                               .apply(extensionLoadingContext));
+  }
+
+  private static class EnricherDelegate {
+
+    final DslResolvingContext dslResolvingContext;
+
+    EnricherDelegate(DslResolvingContext dslResolvingContext) {
+      this.dslResolvingContext = dslResolvingContext;
+    }
+
+    public void apply(ExtensionLoadingContext extensionLoadingContext) {
+      new IdempotentDeclarationWalker() {
+
+        @Override
+        protected void onConfiguration(ConfigurationDeclaration declaration) {
+          walkDeclaration(declaration.getModelProperty(GlobalElementComponentModelModelProperty.class),
+                          declaration.getAllParameters());
+        }
+
+        @Override
+        protected void onConnectionProvider(ConnectedDeclaration owner,
+                                            ConnectionProviderDeclaration declaration) {
+          if (owner instanceof ConfigurationDeclaration) {
+            walkDeclaration(((ConfigurationDeclaration) owner)
+                .getModelProperty(GlobalElementComponentModelModelProperty.class),
+                            declaration.getAllParameters());
+          }
+        }
+
+        private void walkDeclaration(Optional<GlobalElementComponentModelModelProperty> globalElementModelProperty,
+                                     List<ParameterDeclaration> allParameters) {
+          globalElementModelProperty.ifPresent(modelProperty -> allParameters.stream()
+              .filter(parameterDeclaration -> parameterDeclaration.getType() instanceof StringType)
+              .forEach(parameterDeclaration -> traverseProperty(globalElementModelProperty.get().getGlobalElements(),
+                                                                parameterDeclaration)));
+        }
+      }.walk(extensionLoadingContext.getExtensionDeclarer().getDeclaration());
+    }
+
+    /**
+     * Given a {@code propertyDeclaration} it will look up all the usages in the global elements of the smart connector where
+     * for each occurrence it will extract the stereotypes of the underlying connector's parameter to copy into the current
+     * declaration.
+     * <br/>
+     * If the result of those finding throws repeated stereotypes, then it will reduce that collection by taking the intersection,
```

Comment:
It will not.
If it's used for two different configurations where, lets say, rely on different stereotypes, the intersection will result in an empty stereotype.

Just in case: I don't want to make it fail as it could break compatibility.